core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -375,12 +378,26 @@ public void ensureClusterStateVersion(long clusterStateVersion) {
     }
 
     @Override
-    public long indexTranslogOperations(
-            List<Translog.Operation> operations, int totalTranslogOps) throws TranslogRecoveryPerformer.BatchOperationException {
+    public long indexTranslogOperations(List<Translog.Operation> operations, int totalTranslogOps) throws MapperException, IOException {
```

Comment:
There is already extensive test coverage for this (e.g. test subclasses of ESIndexLevelReplicationTestCase).